modules/react/select/lib/hooks/useSelectInput.ts

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import {composeHooks, createElemPropsHook} from '@workday/canvas-kit-react/common';
+import {useComboboxInput, useScrollToIndexOnClick} from '@workday/canvas-kit-react/combobox';
+import {useSelectModel} from './useSelectModel';
+
+/**
+ * `useSelectInput` extends {@link useComboboxInput useComboboxInput}  and adds type ahead functionality and select specific [keyboard support](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/).
+ */
+export const useSelectInput = composeHooks(
+  useComboboxInput,
+  useScrollToIndexOnClick,
+  createElemPropsHook(useSelectModel)((model, ref) => {
+    const keySofar = React.useRef('');
+    const timer = React.useRef<ReturnType<typeof setTimeout>>();
+
+    // Reset after each keystroke to support type ahead
+    const startClearKeysSoFarTimer = () => {
+      if (timer.current) {
+        clearTimeout(timer.current);
+      }
+      timer.current = setTimeout(() => {
+        keySofar.current = '';
+      }, 500);
+    };
+
+    // Based on the the key typed, this will try and find an index in the array where it matches the string
+    // so if a user types `de` it will try and find an index in the array whose id matches `de` like denver
+    const getIndexByStartString = (
+      startIndex: number,
+      startString: string,
+      endIndex: number = model.state.items.length,
+      ignoreDisabled: boolean = true
+    ): number => {
+      for (let i = startIndex; i < endIndex; i++) {
+        const label = model.state.items[i].id.toLowerCase();
+
+        if (label.indexOf(startString.toLowerCase()) === 0) {
+          if (
+            !ignoreDisabled ||
+            (ignoreDisabled && !model.state.nonInteractiveIds.includes(model.state.items[i].id))
+          ) {
+            return i;
+          }
+        }
+      }
+      return -1;
+    };
+    // just use the index you dummy squid
+    const currentItem = model.navigation.getItem(model.state.cursorId, model);
+    const cursorFocusedIndex = model.state.items.findIndex(item => item.id === currentItem.id);
+
+    const handleKeyboardTypeAhead = (key: string, numOptions: number) => {
+      // If the starting point is beyond the list of options, reset it
+      // to the beginning of the list
+      let start = keySofar.current.length === 0 ? cursorFocusedIndex + 1 : cursorFocusedIndex;
+
+      start = start === numOptions ? 0 : start;
```

Comment:
```suggestion
      const number = keySofar.current.length === 0 ? cursorFocusedIndex + 1 : cursorFocusedIndex;
      const start = number === numOptions ? 0 : number;
```
Maybe you can do not have changeable var. Do you reassign this var in different place?